src/services/download/sentry.rs

```diff
@@ -219,37 +219,60 @@ impl SentryDownloader {
         file_source: SentryObjectFileSource,
         destination: PathBuf,
     ) -> Result<DownloadStatus, DownloadError> {
-        let download_url = file_source.url();
-        log::debug!("Fetching debug file from {}", download_url);
-        let result = future_utils::retry(|| {
-            self.client
-                .get(download_url.clone())
-                .header("User-Agent", USER_AGENT)
-                .header(
-                    "Authorization",
-                    format!("Bearer {}", &file_source.source.token),
-                )
-                .send()
-        });
-
-        match result.await {
+        match future_utils::retry(|| {
+            self.download_source_once(file_source.clone(), destination.clone())
+        })
+        .await
+        {
+            Ok(status) => {
+                log::debug!(
+                    "Fetched debug file from {}: {:?}",
+                    file_source.url(),
+                    status
+                );
+                Ok(status)
+            }
+            Err(err) => {
+                log::debug!(
+                    "Failed to fetch debug file from {}: {}",
+                    file_source.url(),
+                    err
+                );
+                Err(err)
+            }
+        }
+    }
+
+    async fn download_source_once(
+        &self,
+        source: SentryObjectFileSource,
+        destination: PathBuf,
+    ) -> Result<DownloadStatus, DownloadError> {
+        match self
+            .client
+            .get(source.url())
+            .header("User-Agent", USER_AGENT)
+            .bearer_auth(&source.source.token)
```

Comment:
This changed from the previous implementation. It looks like it's not a big deal, but lets make sure that this doesn't ship an unintended change in behavior.

Reply:
Yes, this uses a feature from reqwest which we didn't use yet.  I switched to it because it seems like a safer API to do the same.  It is something to watch during deployment though to make sure there's no regression (I do recall checking the code though and it seemed fine).